### PR TITLE
core/msg.c: fixed msg_queue_print()

### PR DESCRIPTION
**Pull request by JulianHolzwarth**
### Contribution description
This PR fixes `msg_queue_print`. When the msg_queue is full the  `msg_queue_print` does not print out the messages.

### Testing procedure

change main in msg_avail to :
```
int main(void)
{
    msg_t msges[MSG_QUEUE_LENGTH];

    msg_init_queue(msg_queue, MSG_QUEUE_LENGTH);

    puts("[START]");

    /* add message to own queue */
    for (int idx = 0; idx < MSG_QUEUE_LENGTH; ++idx) {
        msges[idx].type = idx;
        msg_send_to_self(msges + idx);
        LOG_INFO("+ add msg: %d\n", idx);
        if (msg_avail() != (idx) + 1) {
            puts("[FAILED]");
            return 1;
        }
    }
    msg_queue_print();
    /* receive available messages in queue */
    for (int idx = msg_avail(); idx > 0; --idx) {
        msg_t msg;
        msg_receive(&msg);
        LOG_INFO("- got msg: %d\n", (MSG_QUEUE_LENGTH - idx));
        if ((int)msg.type != (MSG_QUEUE_LENGTH - idx)
            || msg_avail() != idx - 1) {
            puts("[FAILED]");
            return 1;
        }
    }
    puts("[SUCCESS]");
    return 0;
}
```

### CODE REVIEW

File: core/msg.c [hunk 1]
```diff
@@ -454,23 +454,27 @@ void msg_init_queue(msg_t *array, int num)
...
-    printf("    size: %u (avail: %d)\n", msg_queue->mask + 1,
-           cib_avail(msg_queue));
+    printf("    size: %u (avail: %d)\n", msg_queue->mask + 1, msg_counter);
 
-    for (; i != (msg_queue->write_count & msg_queue->mask);
-         i = (i + 1) & msg_queue->mask) {
-        msg_t *m = &msg_array[i];
+    for ( int i = 0; i < msg_counter; i++)
+    {
+        msg_t *m = &msg_array[first_msg+i];
```
Comment (kaspar030) [inline core/msg.c:474]: doesn't this read after the end of msg_array? e.g., with a queue size of 8, `msg_counter > 1` and `first_msg==queue_size-1`.

Comment (JulianHolzwarth) [reply core/msg.c:474]: yes thank you i missed that. it should be  ` msg_t *m = &msg_array[(first_msg+i)% (msg_queue->mask + 1)];`
right?

Comment (kaspar030) [reply core/msg.c:474]: or just `msg_t *m = &msg_array[(first_msg+i) & msg_queue->mask];`, the size is power of two.

Comment (kaspar030) [reply core/msg.c]: using `&` saves a division...

Comment (JulianHolzwarth) [reply core/msg.c]: ok thanks.

Comment (JulianHolzwarth) [reply core/msg.c]: can i rebase? (fixup)

Comment (kaspar030) [reply core/msg.c]: sure!

Comment (JulianHolzwarth) [reply core/msg.c]: ok

**Review by benpicco — DISMISSED** [review]

**Comment by benpicco** [comment]
Can you also provide a testcase for this?
Right now `msg_queue_print()` is not called by any code in RIOT.

**Comment by JulianHolzwarth** [comment]
> Can you also provide a testcase for this?
> Right now `msg_queue_print()` is not called by any code in RIOT.

Yes. I will do that.

**Review by benpicco — APPROVED** [review]

**Comment by benpicco** [comment]
You will need to rebase to trigger CI.

**Comment by JulianHolzwarth** [comment]
I rebase and added a new test for `msg_queue_print()`.

**Comment by JulianHolzwarth** [comment]
fixed whitespace

**Comment by fjmolinas** [comment]
All green here :)

**Comment by miri64** [comment]
> All green here :)

Except uncrustify wants some changes ;-).

**Review by miri64 — APPROVED** [review]
ACK! Let's finally merge this!

**Comment by fjmolinas** [comment]
Seems like the test is failing on hw https://ci.riot-os.org/RIOT-OS/RIOT/master/5d7477fa5f240e7989acb1158c8342d06b9c8360/output.html.

File: tests/msg_queue_print/tests/01-run.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect("No message queue")
+    child.expect("Message queue of thread 2")
+    child.expect_exact('size: 8 (avail: 0)')
+    child.expect("Message queue of thread 2")
```
Comment (fjmolinas) [inline tests/msg_queue_print/tests/01-run.py:19]: Thread number should not be hardcoded I think..

File: tests/msg_queue_print/tests/01-run.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    child.expect_exact('size: 8 (avail: 8)')
+    child.expect_exact('type: 0x0000, content: 0 ((nil))')
```
Comment (fjmolinas) [inline tests/msg_queue_print/tests/01-run.py:21]: This is printed as `0` and not `nil` on hardware.

**Pull request merged**